dev/bots/prepare_package.dart

```diff
@@ -426,6 +428,47 @@ class ArchiveCreator {
     await _unzipArchive(gitFile, workingDirectory: minGitPath);
   }
 
+  /// Downloads an archive of every package that is present in the temporary
+  /// pub-cache from pub.dev. Stores the archives in
+  /// $flutterRoot/.pub-preload-cache.
+  ///
+  /// These archives will be installed in the user-level cache on first
+  /// following flutter command that accesses the cache.
+  ///
+  /// This assumes that all packages currently in the cache are installed from
+  /// pub.dev.
+  Future<void> _downloadPubPackageArchives() async {
+    final Pool pool = Pool(10); // Number of simultaneous downloads.
+    final http.Client client = http_retry.RetryClient(http.Client());
```

Comment:
nit: this [won't retry](https://github.com/dart-lang/http/blob/master/pkgs/http/lib/retry.dart#L104-L128) if the response fails, say the TCP connection breaks while fetching bytes -- as far as I can see it only retries requests.

Reply:
I now added full retries